xmodule/util/keys.py

```diff
@@ -28,6 +27,14 @@ class BlockKey(NamedTuple):
     def from_usage_key(cls, usage_key):
         return cls(usage_key.block_type, usage_key.block_id)
 
+    def __str__(self) -> str:
+        return f"{self.type}:{self.id}"
+
+    @classmethod
+    def from_string(cls, s: str) -> Self:
+        parts = s.split(':')
```

Comment:
for parsing robustness, can we throw a ValueError here if there are not exactly two string parts, both non-empty?